embuilder.py

```diff
@@ -15,9 +15,13 @@
 import argparse
 import logging
 import sys
+import os
 import time
 from contextlib import contextmanager
 
+__rootpath__ = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(__rootpath__)
```

Comment:
Do you know why these changes would be needed?   Isn't python supposed to include the script directory implicitly?   What version of python are you using here?

Reply:
It's also a mystery to me that I am currently trying to solve. For some reason, the Python interpreter obtained from `emsdk install latest` is ignoring `PYTHONPATH` variable (even when `-E` argument is removed from bat files) and it does not contain current working dir in `sys.path` by default.

Reply:
But you shouldn't need to set `PYTHONPATH` at all should you?   

The current working directory is not normally part of `sys.path` but the script directory certainly should be.

e.g.:

```
$ cat emscripten/test.py 
import sys
print(sys.path)
$ pwd
/usr/local/google/home/sbc/dev/wasm
$ python3 emscripten/test.py
['/usr/local/google/home/sbc/dev/wasm/emscripten', '/usr/local/buildtools/current/sitecustomize', '/usr/lib/python310.zip', '/usr/lib/python3.10', '/usr/lib/python3.10/lib-dynload', '/usr/local/google/home/sbc/.local/lib/python3.10/site-packages', '/usr/local/lib/python3.10/dist-packages', '/usr/lib/python3/dist-packages', '/usr/lib/python3.10/dist-packages']
```

Note that `pwd` does not appear in the path, but that script directory does.

Is there some reason you want to remove the `-E` from the python command?

Reply:
Your are right in every point. I am just not sure what you mean by "script directory"? `site-packages`? Nevertheless, currently I just need to find a simple workaround to continue working on making unit tests working on Arm64 with Visual Studio 2022 and I'll try to figure out what is wrong with the Python interpreted acquired by `emdsk.py` later. With the system-wide Python-interpreter installed from EXE installer, the `runner.py` is loading the Python local scripts just fine.

Reply:
@sbc100, JFYI, the workarounds for `sys.path` imports are related to this Python issue python/cpython#101867

Reply:
We intentionally don't use the `embed` versions of python for this (and perhaps other) reasons.

How are you installing python?  Why not just use the distro package of python?  this is what we normally use/recommend on linux.  (Note that while the mac and windows versions of emsdk install python themselves the linux version expects you to have it installed already).

Reply:
Oh I see, you are trying to install on windows, not linux.  See #16466 for when we switched to the nuget packages.

Reply:
See `scripts/update_python.py` for how to build the patched nuget installer.   I think you would need to modify that script to support non x86_64 arch.

Reply:
> See `scripts/update_python.py` for how to build the patched nuget installer. I think you would need to modify that script to support non x86\_64 arch.

Great, thank you. That's actually what we are working on.